File: src/fmj/magic/MagicAuxiliary.kt
package fmj.magic

import fmj.characters.FightingCharacter
import kotlin.math.min

/**
 * 04辅助型
 * @author Chen
 */
class MagicAuxiliary : BaseMagic() {

    private var mHp: Int = 0//0~100，表示被施展者恢复生命的百分比（起死回生）

    override fun setOtherData(buf: ByteArray, offset: Int) {
        mHp = get2BytesInt(buf, offset + 0x12)
    }

    override fun use(src: FightingCharacter, dst: FightingCharacter) {
        dst.hp = min(dst.maxHP, dst.hp + dst.maxMP * mHp / 100)
    }
}

File: packages/core/src/magic/magic-auxiliary.ts
import { readUint16 } from '@/lib/resource-utils';
import { BaseMagic } from './base-magic';

export class MagicAuxiliary extends BaseMagic {
  hpPercent = 0;

  protected setOtherData(buf: Uint8Array, offset: number): void {
    this.hpPercent = readUint16(buf, offset + 0x12);
  }
}
